### ISSUE TITLE
Unknown command and not functionality after fresh install

### ISSUE THREAD

**Issue by bentolor** [oh-my-fish/oh-my-fish#321]
Today I updates fish and tried to do the same for OMF. After installing OMF no `omf` command works. See the protocoll of installation and usage below.

Any ideas what is going wrong?

# Installation
````
$ curl -L https://github.com/oh-my-fish/oh-my-fish/raw/master/bin/install > install
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   140  100   140    0     0    283      0 --:--:-- --:--:-- --:--:--   283
100  4654  100  4654    0     0   7418      0 --:--:-- --:--:-- --:--:--  7418
$ fish install
Installing Oh My Fish to /home/ben/.local/share/omf...
Cloning master from https://github.com/oh-my-fish/oh-my-fish.git...
Existent config.fish found at /home/ben/.config/fish
↳ Moving to /home/ben/.config/fish/config.1464005872.copy
Existent fish_prompt.fish found at /home/ben/.config/fish/functions
↳ Moving to /home/ben/.config/fish/functions/fish_prompt.1464005872.copy
Adding startup code to fish config file...
Building Oh My Fish configuration...
No matches for wildcard '{$OMF_CONFIG,$OMF_PATH}/themes/*'.
~/.local/share/omf/pkg/omf/functions/packages/omf.packages.list.fish (line 1): basename -a {$OMF_CONFIG,$OMF_PATH}/themes/* 2>/dev/null
                                                                                           ^
in command substitution
	called on line 51 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.list.fish

in function '__omf.packages.list.installed'
	called on line 73 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.list.fish

in function 'omf.packages.list'
	called on line 2 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish
	with parameter list '--installed'

in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish

in function 'omf.bundle.install'
	called on line 6 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.install.fish

in function 'omf.cli.install'
	called on line 47 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'install'

fish: Unknown command 'omf::dim'
~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish (line 1): omf::dim
                                                                                  ^
in command substitution
	called on line 13 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish

in function 'omf.packages.install'
	called on line 22 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish
	with parameter list 'default'

in function 'omf.bundle.install'
	called on line 6 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.install.fish

in function 'omf.cli.install'
	called on line 47 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'install'

fish: Unknown command 'omf::off'
~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish (line 1): omf::off
                                                                                  ^
in command substitution
	called on line 13 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish

in function 'omf.packages.install'
	called on line 22 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish
	with parameter list 'default'

in function 'omf.bundle.install'
	called on line 6 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.install.fish

in function 'omf.cli.install'
	called on line 47 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'install'


fish: Unknown command 'omf::em'
~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish (line 1): omf::em
                                                                                  ^
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish

in function '__omf.packages.install.success'
	called on line 44 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish
	with parameter list 'theme default'

in function 'omf.packages.install'
	called on line 22 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish
	with parameter list 'default'

in function 'omf.bundle.install'
	called on line 6 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.install.fish

in function 'omf.cli.install'
	called on line 47 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'install'

fish: Unknown command 'omf::off'
~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish (line 1): omf::off
                                                                                  ^
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish

in function '__omf.packages.install.success'
	called on line 44 of file ~/.local/share/omf/pkg/omf/functions/packages/omf.packages.install.fish
	with parameter list 'theme default'

in function 'omf.packages.install'
	called on line 22 of file ~/.local/share/omf/pkg/omf/functions/bundle/omf.bundle.install.fish
	with parameter list 'default'

in function 'omf.bundle.install'
	called on line 6 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.install.fish

in function 'omf.cli.install'
	called on line 47 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'install'


Installation successful!
Willkommen zu fish, der freundlichen interaktiven Shell
Anweisungen zur fish-Benutzung erhalten Sie �ber help
⋊> ~ 
````

# Usage

````
⋊> ~ omf help                                                                                                                                                                                   14:17:54
omf::dim: command not found
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.help.fish

in function 'omf.cli.help'
	called on line 29 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'help'

omf::off: command not found
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.help.fish

in function 'omf.cli.help'
	called on line 29 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'help'

omf::em: command not found
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.help.fish

in function 'omf.cli.help'
	called on line 29 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'help'

omf::off: command not found
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.help.fish

in function 'omf.cli.help'
	called on line 29 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

in function 'omf'
	called on standard input
	with parameter list 'help'

omf::em: command not found
in command substitution
	called on line 0 of file ~/.local/share/omf/pkg/omf/functions/cli/omf.cli.help.fish

in function 'omf.cli.help'
	called on line 29 of file ~/.local/share/omf/pkg/omf/functions/omf.fish

   <Many, man, many lines later>

⋊> ~    
````

**Comment by bobthecow** [comment]
what does `echo $OMF_CONFIG; echo $OMF_PATH` say?

**Comment by bentolor** [comment]
Before install:
````
~> echo $OMF_CONFIG; echo $OMF_PATH
````

After install:
````
⋊> ~ echo $OMF_CONFIG; echo $OMF_PATH                                                                                                                                                           14:05:49
/home/ben/.config/omf
/home/ben/.local/share/omf
```

Both locations look fine:

````
⋊> ~/.c/omf cat bundle                                                                                                                                                                          14:07:31
theme default
⋊> ~/.c/omf cat theme                                                                                                                                                                           14:07:32
default
````
and
````
⋊> ~/.l/s/omf on master ◦ git status                                                                                                                                                            14:09:59
Auf Branch master
Ihr Branch ist auf dem selben Stand wie 'origin/master'.
nichts zu committen, Arbeitsverzeichnis unverändert
````

**Comment by oranja** [comment]
@bentolor, please try
```fish
mkdir -p $OMF_PATH/themes
```
and then try to reinstall.

I couldn't yet figure out what was the change that introduced this bug, but I have a theory for why this is happening.
The relevant flow is: `cli.install` -> `bundle.install` -> `packages.list`.

1. `cli.install` calls `bundle.install` in order to get the default theme cloned and set. It does not create a `themes` folder because this should happen automatically with the first theme installed.
2. `bundle.install`, in turn, calls `packages.list` to see what is already installed and what's missing (IIRC).
3. `pacakge.list`, unfortunately, expects the `themes` folder to already exist, or otherwise, you get this angry fish failing to expand the wildcards match:

  > No matches for wildcard '{$OMF_CONFIG,$OMF_PATH}/themes/*'

And there you have it.

Unless anyone beats me to it, I will find out when was this bug introduced and propose a fix later today.

**Comment by bentolor** [comment]
Thanks @oranja. This sound reasonable. 

The workaround does not work though, as the script checks at startup. So this becomes a sort of an chicken & egg problem.

````
~> mkdir -p /home/ben/.local/share/omf/themes
~> fish install
Installing Oh My Fish to /home/ben/.local/share/omf...
Aborting: Existing installation detected.
````

**Comment by oranja** [comment]
Mmmm. Busted.
Well then, next two options are either wait for a fix, or remove the folder
and tweak the install script as a workaround (just move the `mkdir` line to
the beginning of the `install_omf` function)

**Comment by derekstavis** [comment]
Fixed in #325
